napari/_vispy/overlays/scale_bar.py

```diff
@@ -12,10 +12,13 @@
 class VispyScaleBarOverlay(ViewerOverlayMixin, VispyCanvasOverlay):
     """Scale bar in world coordinates."""
 
-    def __init__(self, *, viewer, overlay, parent=None) -> None:
+    def __init__(
+        self, *, viewer, overlay, bg_color_override, parent=None
+    ) -> None:
         self._target_length = 150
         self._scale = 1
         self._unit = None
+        self.bg_color_override = bg_color_override
```

Comment:
self.node.parent.parent.canvas actually points to scene_canvas and not the VispyCanvas. Since background_color_override is a property of VispyCanvas this property is not accessible anymore to VispyScaleBarOverlay.

Reply:
Just to note, changing the canvas background color does not change the background_color_override property. This will stay none.

Reply:
Yeah I think canvas color independent of theme is still a good idea and ideally something to set in GUI at some point, but less crucial with https://github.com/napari/napari/pull/5487 ?
By default, canvas color could be set by theme, but it should be able to be changed.
I can see people wanting a neutral grey canvas regardless of preferred UI theme.

Reply:
I got a solution I think which I will put in 2 commits. This should also be more future proof when one starting thinking of multichannel and wanting an override for 1 viewbox (showing HE and fluorescent imaging at the same time, one white background and 1 dark). VispyCanvas.view is actually a viewbox for which the bgcolor can be accessed in VispyScaleBarOverlay by doing self.node.parent.bgcolor. This bgcolor is already equal to VispyCanvas.background_override_color if this has been set, or theme color otherwise.

Reply:
@brisvag @psobolewskiPhD [c375a02](https://github.com/napari/napari/pull/5432/commits/c375a02165f7515829e3d1dd961ed730a397b905) seems to be the solution for now. It keeps background_color_override while not changing the overlay API. 

I did also change bg_color_override so that it now sets the bgcolor of self.view. [c99827e](https://github.com/napari/napari/pull/5432/commits/c99827e535502a066333428692c98faa83d7da80) This if statement:
```
if self.view in self.central_widget._widgets:
```
already takes into account that for multichannel view, self.view is not in the widgets. In this case you would have a grid with multiple viewboxes, for which each one should have the possibility of overriding color. One particular way of doing this would be to first have Napari Qt directly taking mouse events, passing position on to Vispy where we would have logic to see over which viewbox a right click happened. If having a single viewbox then set the bg color of that viewbox and otherwise call a function to check for which viewbox the bgcolor needs to be changed.

What do you think? Is this sufficient in its current state for this PR?